rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cerrno>
+#include <memory>
+#include <mutex>
+#include <inttypes.h>
+
+#include <rcutils/allocator.h>
+#include <rcutils/filesystem.h>
+#include <rcutils/get_env.h>
+#include <rcutils/process.h>
+#include <rcutils/snprintf.h>
+#include <rcutils/time.h>
+
+#include "spdlog/spdlog.h"
+#include "spdlog/sinks/basic_file_sink.h"
+
+#include "rcl_logging_spdlog/logging_interface.h"
+
+#define RC_LOGGING_RET_OK                          (0)
+#define RC_LOGGING_RET_ERROR                       (2)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+static std::mutex g_logger_mutex;
+static std::shared_ptr<spdlog::logger> g_root_logger = nullptr;
+
+/* These are defined here to match the severity levels in rcl. They provide a consistent way for external logger
+    implementations to map between the incoming integer severity from ROS to the concept of DEBUG, INFO, WARN, ERROR,
+    and FATAL*/
+enum RC_LOGGING_LOG_SEVERITY
```

Comment:
Could we use the rcutils log levels instead of redefining here?

https://github.com/ros2/rcutils/blob/54ecb72a3784cc54d361f38bb9cab5bad7172772/include/rcutils/logging.h#L164

Reply:
Oh, good call.  Will change to that.